### ISSUE TITLE
Missing fallback support while signing out a device

### ISSUE THREAD

**Issue by waclaw66** [devture/matrix-synapse-shared-secret-auth#12]
Please see https://github.com/vector-im/element-web/issues/20292

**Comment by Wunderharke** [comment]
See vector-im/element-web#19605.

**Comment by spantaleev** [comment]
No upstream progress on these issues, unfortunately.

**Comment by timokoesters** [comment]
> Interactive auth chooses the first flow it sees

Is it possible to change the order such that shared secret auth is not the first one?

**Comment by spantaleev** [comment]
Good idea! It only makes sense if both supported login types (`m.login.password` and `com.devture.shared_secret_auth`) are enabled. We can then play around with the order and try to register `m.login.password` first.

If only `com.devture.shared_secret_auth` is registered, then it's not applicable.

However.. testing on Synapse v1.53.0, https://github.com/vector-im/element-web/issues/19605 is no longer an issue, because `m.login.password` is always first in the list regardless of ordering and regardless of whether shared-secret-auth enables the `m.login.password` login type. Looks like this has become a non-issue now - due to an intentional fix on the Synapse side or due to some random change. I've reported this [here](https://github.com/vector-im/element-web/issues/19605#issuecomment-1053410778)

**Comment by Sharparam** [comment]
`m.login.password` is not listed first on my Synapse v1.55.0 instance, so this is still an issue. Perhaps some version after v1.53.0 changed the ordering again.

Edit: Response after Element calls `/_matrix/client/r0/delete_devices` when trying to remove an old device:

```json
{
  "session": "<snip>",
  "flows" : [
    {
      "stages": [ "com.devture.shared_secret_auth" ]
    },
    {
      "stages": [ "m.login.password" ]
    }
  ],
  "params": {}
}
```
